### ISSUE TITLE
Directly loading image url with blocked media does not allow zooming in via LMB

### ISSUE THREAD

**Issue by xftroxgpx** [gorhill/uMatrix#725]
Hello.

1. load an image url directly in chromium's address bar, like:
https://i.abcdn.co/ff54%20gcc6.3%20patch%20test.png
2. assuming display resolution is too small for the image to be displayed full size in browser, then the mouse cursor turns into a magnifier with a `+` sign, allowing LMB click on the image to zoom into it.
3. ensure Media in uMatrix is blocked, then LMB on the loaded image will not do anything, it will not zoom the image.
4. ensure Media in Matrix is green/allowed, **EDIT:** reload page(image), then LMB on the image will zoom it.

screens:

cannot zoom image:
![screenshot_2017-03-08_07-48-01](https://cloud.githubusercontent.com/assets/26243389/23693459/e66ae8f0-03d4-11e7-8bb1-df0994ee80cd.png)

can zoom image:
![screenshot_2017-03-08_07-49-02](https://cloud.githubusercontent.com/assets/26243389/23693468/ebc4e79c-03d4-11e7-84d1-ecd56b9147a0.png)

I could provide more elaborate steps(if needed) but I didn't want them to be noise, since this seems easily reproducible on any chromium config(presumably).
Chromium	59.0.3035.0
OS	Linux
JavaScript	V8 5.9.26

Is there anything uMatrix can do to workaround this?
Thanks.

**Comment by Atavic** [comment]
That's working correctly. By zooming in you can trigger onclick events in web pages.

**Comment by xftroxgpx** [comment]
But as far as I know, @Atavic, (and I don't know much, I'll give you that) the zoom is only allowed when the image is standalone(image url in address bar); otherwise the webpage has to provide the zoom in functionality itself through javascript(hopefully that's the only way, eg. not possible via css-only; i presume).

Here's a question:
Can `Content-Type: image/`*  be interpreted by the browser as a webpage or html perhaps? Or, if not, then maybe that(+some more) can be used to auto-allow  images from 1st party, thus allowing the zoom.

In other words, can the server serve a html and lie that it's an `image` and the browser would then attempt to show it as a webpage/run_javascript?

"+some more" would be: there needs to be a way to see that the request for the image is the same as the url in the address bar, otherwise I presume there's no way to differentiate between the image being standalone(in the url address bar) or part of the webpage resources being loaded.

And Would the above logic still trigger if the image is alone in a frame or iframe ? then what @Atavic said makes sense, presuming javascript in other frame(s) can access events in others(?). Presumably there would be possible to differentiate between url in address bar and frames since 'frame' is a column in uMatrix.

<details>
<pre>
$ curl -I https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png
HTTP/1.1 200 OK
Cache-Control: max-age=2592000
Last-Modified: Thu, 27 Apr 2017 10:11:47 GMT
ETag: "c4327bb26aa856f783346c55366dc944"
Content-Type: image/png
Content-Length: 447268
Accept-Ranges: bytes
Date: Thu, 27 Apr 2017 10:24:21 GMT
Via: 1.1 varnish
Age: 752
Connection: keep-alive
X-Served-By: cache-fra1225-FRA
X-Cache: HIT
X-Cache-Hits: 1
X-Fastly-Request-ID: d39166c1d1fea3aeed5c8d5724d29c55a733a45a
Server: GitHub Cloud
Timing-Allow-Origin: https://github.com
</pre>
</details>

**Comment by xftroxgpx** [comment]
In a way, you could argue that, since nothing is allowed in uMatrix, then even though the image itself is in the address bar, uMatrix shouldn't display it at all xD

There might be something else going on, because I vaguely remember images being loaded/displayed fast, then hidden, in some webpages... which made me think chromium is really loading them anyway even though uMatrix tells it not to. I'll investigate to see if this is so, and if not, edit this with strike.

**Comment by xftroxgpx** [comment]
re previous comment first paragraph: the image goes away after loading `Developer Tools` Ctrl+Shift+J

Steps:
0. uMatrix must have block all (at least for cloud.githubusercontent.com )
`* * * block`
1. put this in url and press enter: https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png
2. observe that image is shown on screen
3. open Developer Tools by pressing Ctrl+Shift+J (j or i, apparently)
4. the image went away, and it's all black now
5. close Developer Tools by pressing the same Ctrl+Shift+J
6. the image is still gone, background is still black-ish.

console shows:
```
13:28:51.650 chrome-extension://iphimjoabppgpifmkfadpbjigcefepdp/js/contentscript.js:90 Uncaught Error: uBlock Origin: aborting content scripts for https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png
    at chrome-extension://iphimjoabppgpifmkfadpbjigcefepdp/js/contentscript.js:90
13:28:51.848 /favicon.ico Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
```

**Comment by xftroxgpx** [comment]
OH, actually the above is just a side-effect of 'resize', aka resizing main window makes the image go away! Maybe chromium realizes(only upon window resize) that it shouldn't have shown the image(because uMatrix blocked it? if it did)

**Comment by xftroxgpx** [comment]
So I guess this issue is: the image shouldn't show up in the first place? that's why zoom isn't working, it's not supposed to work because the image isn't supposed to show up either. My guess.

**Comment by xftroxgpx** [comment]
Here be interesting chain of events(?):

<details>
<pre>
1723: URL_REQUEST
https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png
Start Time: 2017-04-27 13:44:26.309

t=  0 [st=  0] +REQUEST_ALIVE  [dt=305]
                --> priority = "HIGHEST"
                --> url = "https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png"
t=  0 [st=  0]    DELEGATE_INFO  [dt=4]
                  --> delegate_blocked_by = "NavigationResourceThrottle"
t=  4 [st=  4]   +URL_REQUEST_DELEGATE  [dt=2]
t=  4 [st=  4]      DELEGATE_INFO  [dt=2]
                    --> delegate_blocked_by = "extension uMatrix"
t=  6 [st=  6]   -URL_REQUEST_DELEGATE
t=  6 [st=  6]    URL_REQUEST_START_JOB  [dt=1]
                  --> load_flags = 37120 (MAIN_FRAME_DEPRECATED | MAYBE_USER_GESTURE | VERIFY_EV_CERT)
                  --> method = "GET"
                  --> url = "https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png"
t=  7 [st=  7]   +URL_REQUEST_START_JOB  [dt=146]
                  --> load_flags = 37120 (MAIN_FRAME_DEPRECATED | MAYBE_USER_GESTURE | VERIFY_EV_CERT)
                  --> method = "GET"
                  --> url = "https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png"
t=  7 [st=  7]     +URL_REQUEST_DELEGATE  [dt=2]
t=  7 [st=  7]        DELEGATE_INFO  [dt=2]
                      --> delegate_blocked_by = "extension uMatrix"
t=  9 [st=  9]     -URL_REQUEST_DELEGATE
t=  9 [st=  9]      HTTP_CACHE_GET_BACKEND  [dt=0]
t=  9 [st=  9]      HTTP_CACHE_OPEN_ENTRY  [dt=4]
                    --> net_error = -2 (ERR_FAILED)
t= 13 [st= 13]      HTTP_CACHE_CREATE_ENTRY  [dt=3]
t= 16 [st= 16]      HTTP_CACHE_ADD_TO_ENTRY  [dt=0]
t= 16 [st= 16]     +HTTP_STREAM_REQUEST  [dt=104]
t= 16 [st= 16]        HTTP_STREAM_JOB_CONTROLLER_BOUND
                      --> source_dependency = 1726 (HTTP_STREAM_JOB_CONTROLLER)
t=120 [st=120]        HTTP_STREAM_REQUEST_BOUND_TO_JOB
                      --> source_dependency = 1727 (HTTP_STREAM_JOB)
t=120 [st=120]     -HTTP_STREAM_REQUEST
t=120 [st=120]     +HTTP_TRANSACTION_SEND_REQUEST  [dt=1]
t=120 [st=120]        HTTP_TRANSACTION_SEND_REQUEST_HEADERS
                      --> GET /assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png HTTP/1.1
                          Host: cloud.githubusercontent.com
                          Connection: keep-alive
                          Upgrade-Insecure-Requests: 1
                          User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3081.0 Safari/537.36
                          Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
                          Accept-Encoding: gzip, deflate, br
                          Accept-Language: en-US,en;q=0.8
t=121 [st=121]     -HTTP_TRANSACTION_SEND_REQUEST
t=121 [st=121]     +HTTP_TRANSACTION_READ_HEADERS  [dt=28]
t=121 [st=121]        HTTP_STREAM_PARSER_READ_HEADERS  [dt=28]
t=149 [st=149]        HTTP_TRANSACTION_READ_RESPONSE_HEADERS
                      --> HTTP/1.1 200 OK
                          Cache-Control: max-age=2592000
                          Last-Modified: Thu, 27 Apr 2017 10:11:47 GMT
                          ETag: "c4327bb26aa856f783346c55366dc944"
                          Content-Type: image/png
                          Content-Length: 447268
                          Accept-Ranges: bytes
                          Date: Thu, 27 Apr 2017 11:44:43 GMT
                          Via: 1.1 varnish
                          Age: 5574
                          Connection: keep-alive
                          X-Served-By: cache-fra1229-FRA
                          X-Cache: HIT
                          X-Cache-Hits: 1
                          X-Fastly-Request-ID: 002d8551c13c909e465cb66028a62efb3839411b
                          Server: GitHub Cloud
                          Timing-Allow-Origin: https://github.com
t=149 [st=149]     -HTTP_TRANSACTION_READ_HEADERS
t=149 [st=149]      HTTP_CACHE_WRITE_INFO  [dt=0]
t=149 [st=149]      HTTP_CACHE_WRITE_DATA  [dt=0]
t=150 [st=150]      HTTP_CACHE_WRITE_INFO  [dt=0]
t=150 [st=150]     +URL_REQUEST_DELEGATE  [dt=3]
t=150 [st=150]        DELEGATE_INFO  [dt=3]
                      --> delegate_blocked_by = "extension uMatrix"
t=153 [st=153]        CHROME_EXTENSION_MODIFIED_HEADERS
                      --> extension_id = "biklejmhpofaebhjdgijbmjhcplnfjec"
t=153 [st=153]     -URL_REQUEST_DELEGATE
t=153 [st=153]   -URL_REQUEST_START_JOB
t=153 [st=153]   +URL_REQUEST_DELEGATE  [dt=1]
t=153 [st=153]      DELEGATE_INFO  [dt=1]
                    --> delegate_blocked_by = "NavigationResourceThrottle"
t=154 [st=154]   -URL_REQUEST_DELEGATE
t=154 [st=154]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=154 [st=154]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=155 [st=155]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 3607
t=155 [st=155]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=155 [st=155]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=155 [st=155]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 6940
t=155 [st=155]    HTTP_TRANSACTION_READ_BODY  [dt=18]
t=173 [st=173]    HTTP_CACHE_WRITE_DATA  [dt=5]
t=178 [st=178]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 21732
t=178 [st=178]    HTTP_TRANSACTION_READ_BODY  [dt=18]
t=196 [st=196]    HTTP_CACHE_WRITE_DATA  [dt=6]
t=202 [st=202]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=202 [st=202]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=202 [st=202]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=203 [st=203]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=203 [st=203]    HTTP_TRANSACTION_READ_BODY  [dt=12]
t=215 [st=215]    HTTP_CACHE_WRITE_DATA  [dt=4]
t=219 [st=219]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=219 [st=219]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=220 [st=220]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=220 [st=220]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=220 [st=220]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=221 [st=221]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=222 [st=222]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=222 [st=222]    HTTP_TRANSACTION_READ_BODY  [dt=6]
t=228 [st=228]    HTTP_CACHE_WRITE_DATA  [dt=2]
t=230 [st=230]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=231 [st=231]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=231 [st=231]    HTTP_CACHE_WRITE_DATA  [dt=5]
t=236 [st=236]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=236 [st=236]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=237 [st=237]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=238 [st=238]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=238 [st=238]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=239 [st=239]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=239 [st=239]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=239 [st=239]    HTTP_TRANSACTION_READ_BODY  [dt=3]
t=242 [st=242]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=243 [st=243]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=243 [st=243]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=244 [st=244]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=244 [st=244]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=244 [st=244]    HTTP_TRANSACTION_READ_BODY  [dt=16]
t=260 [st=260]    HTTP_CACHE_WRITE_DATA  [dt=6]
t=266 [st=266]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=266 [st=266]    HTTP_TRANSACTION_READ_BODY  [dt=11]
t=277 [st=277]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=278 [st=278]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=278 [st=278]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=279 [st=279]    HTTP_CACHE_WRITE_DATA  [dt=4]
t=283 [st=283]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=284 [st=284]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=285 [st=285]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=285 [st=285]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=286 [st=286]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=286 [st=286]    HTTP_CACHE_WRITE_DATA  [dt=6]
t=292 [st=292]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=292 [st=292]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=293 [st=293]    HTTP_CACHE_WRITE_DATA  [dt=4]
t=297 [st=297]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=297 [st=297]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=298 [st=298]    HTTP_CACHE_WRITE_DATA  [dt=3]
t=301 [st=301]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=301 [st=301]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=302 [st=302]    HTTP_CACHE_WRITE_DATA  [dt=2]
t=304 [st=304]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 21773
t=304 [st=304]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=304 [st=304]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=305 [st=305] -REQUEST_ALIVE
</pre>
</details>

**Comment by xftroxgpx** [comment]
I mean, it does `--> delegate_blocked_by = "extension uMatrix"` and the later goes and fetches it anyway? that bastard! :))) chromium i mean =)  (ok, i probably missing something here hehe)

**Comment by Atavic** [comment]
Maybe it's the interaction Chromium/uMatrix the culprit here?

**Comment by xftroxgpx** [comment]
@Atavic I would try to block it via uBlock Origin, but it doesn't [do this kind of blocking](https://github.com/chrisaljoudi/uBlock/issues/456#issuecomment-297686643), apparently. 
(with an empty Whitelist and My rules being only `* * * block` and enabled `I am an advanced user`)

Request for the image isn't blocked:
<details>
<pre>
3589: URL_REQUEST
https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png
Start Time: 2017-04-27 20:19:12.960

t=  0 [st=  0] +REQUEST_ALIVE  [dt=305]
                --> priority = "HIGHEST"
                --> url = "https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png"
t=  0 [st=  0]    DELEGATE_INFO  [dt=2]
                  --> delegate_blocked_by = "NavigationResourceThrottle"
t=  2 [st=  2]   +URL_REQUEST_DELEGATE  [dt=7]
t=  2 [st=  2]      DELEGATE_INFO  [dt=7]
                    --> delegate_blocked_by = "extension uBlock Origin"
t=  9 [st=  9]   -URL_REQUEST_DELEGATE
t=  9 [st=  9]    URL_REQUEST_START_JOB  [dt=0]
                  --> load_flags = 37120 (MAIN_FRAME_DEPRECATED | MAYBE_USER_GESTURE | VERIFY_EV_CERT)
                  --> method = "GET"
                  --> url = "https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png"
t=  9 [st=  9]   +URL_REQUEST_START_JOB  [dt=137]
                  --> load_flags = 37120 (MAIN_FRAME_DEPRECATED | MAYBE_USER_GESTURE | VERIFY_EV_CERT)
                  --> method = "GET"
                  --> url = "https://cloud.githubusercontent.com/assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png"
t=  9 [st=  9]      URL_REQUEST_DELEGATE  [dt=0]
t= 10 [st= 10]      HTTP_CACHE_GET_BACKEND  [dt=0]
t= 10 [st= 10]      HTTP_CACHE_OPEN_ENTRY  [dt=0]
                    --> net_error = -2 (ERR_FAILED)
t= 10 [st= 10]      HTTP_CACHE_CREATE_ENTRY  [dt=0]
t= 10 [st= 10]      HTTP_CACHE_ADD_TO_ENTRY  [dt=0]
t= 10 [st= 10]     +HTTP_STREAM_REQUEST  [dt=105]
t= 10 [st= 10]        HTTP_STREAM_JOB_CONTROLLER_BOUND
                      --> source_dependency = 3592 (HTTP_STREAM_JOB_CONTROLLER)
t=115 [st=115]        HTTP_STREAM_REQUEST_BOUND_TO_JOB
                      --> source_dependency = 3593 (HTTP_STREAM_JOB)
t=115 [st=115]     -HTTP_STREAM_REQUEST
t=115 [st=115]     +HTTP_TRANSACTION_SEND_REQUEST  [dt=1]
t=115 [st=115]        HTTP_TRANSACTION_SEND_REQUEST_HEADERS
                      --> GET /assets/26243389/25478764/aeb3f0e2-2b42-11e7-8f6d-62bc8acac28e.png HTTP/1.1
                          Host: cloud.githubusercontent.com
                          Connection: keep-alive
                          Upgrade-Insecure-Requests: 1
                          User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3081.0 Safari/537.36
                          Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
                          Accept-Encoding: gzip, deflate, br
                          Accept-Language: en-US,en;q=0.8
t=116 [st=116]     -HTTP_TRANSACTION_SEND_REQUEST
t=116 [st=116]     +HTTP_TRANSACTION_READ_HEADERS  [dt=25]
t=116 [st=116]        HTTP_STREAM_PARSER_READ_HEADERS  [dt=25]
t=141 [st=141]        HTTP_TRANSACTION_READ_RESPONSE_HEADERS
                      --> HTTP/1.1 200 OK
                          Cache-Control: max-age=2592000
                          Last-Modified: Thu, 27 Apr 2017 10:11:47 GMT
                          ETag: "c4327bb26aa856f783346c55366dc944"
                          Content-Type: image/png
                          Content-Length: 447268
                          Accept-Ranges: bytes
                          Date: Thu, 27 Apr 2017 18:19:30 GMT
                          Via: 1.1 varnish
                          Age: 29261
                          Connection: keep-alive
                          X-Served-By: cache-fra1240-FRA
                          X-Cache: HIT
                          X-Cache-Hits: 1
                          X-Fastly-Request-ID: 3f0b37315c671bd7eeba7c81fe965ec271b988a4
                          Server: GitHub Cloud
                          Timing-Allow-Origin: https://github.com
t=141 [st=141]     -HTTP_TRANSACTION_READ_HEADERS
t=141 [st=141]      HTTP_CACHE_WRITE_INFO  [dt=1]
t=142 [st=142]      HTTP_CACHE_WRITE_DATA  [dt=0]
t=142 [st=142]      HTTP_CACHE_WRITE_INFO  [dt=0]
t=142 [st=142]     +URL_REQUEST_DELEGATE  [dt=4]
t=142 [st=142]        DELEGATE_INFO  [dt=4]
                      --> delegate_blocked_by = "extension uBlock Origin"
t=146 [st=146]        CHROME_EXTENSION_MODIFIED_HEADERS
                      --> extension_id = "iphimjoabppgpifmkfadpbjigcefepdp"
t=146 [st=146]     -URL_REQUEST_DELEGATE
t=146 [st=146]   -URL_REQUEST_START_JOB
t=146 [st=146]   +URL_REQUEST_DELEGATE  [dt=1]
t=146 [st=146]      DELEGATE_INFO  [dt=1]
                    --> delegate_blocked_by = "NavigationResourceThrottle"
t=147 [st=147]   -URL_REQUEST_DELEGATE
t=147 [st=147]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=147 [st=147]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=147 [st=147]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 2269
t=147 [st=147]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=148 [st=148]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=148 [st=148]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 8277
t=148 [st=148]    HTTP_TRANSACTION_READ_BODY  [dt=18]
t=166 [st=166]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=166 [st=166]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 19143
t=166 [st=166]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=166 [st=166]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=167 [st=167]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 2589
t=167 [st=167]    HTTP_TRANSACTION_READ_BODY  [dt=20]
t=187 [st=187]    HTTP_CACHE_WRITE_DATA  [dt=3]
t=190 [st=190]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=190 [st=190]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=191 [st=191]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=191 [st=191]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=191 [st=191]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=192 [st=192]    HTTP_CACHE_WRITE_DATA  [dt=2]
t=194 [st=194]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=194 [st=194]    HTTP_TRANSACTION_READ_BODY  [dt=14]
t=208 [st=208]    HTTP_CACHE_WRITE_DATA  [dt=4]
t=212 [st=212]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=212 [st=212]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=213 [st=213]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=214 [st=214]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=214 [st=214]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=214 [st=214]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=215 [st=215]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=215 [st=215]    HTTP_TRANSACTION_READ_BODY  [dt=3]
t=218 [st=218]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=219 [st=219]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=219 [st=219]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=220 [st=220]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=220 [st=220]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=220 [st=220]    HTTP_TRANSACTION_READ_BODY  [dt=59]
t=279 [st=279]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=280 [st=280]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 16384
t=280 [st=280]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=281 [st=281]    HTTP_CACHE_WRITE_DATA  [dt=4]
t=285 [st=285]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=285 [st=285]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=286 [st=286]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=286 [st=286]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=286 [st=286]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=287 [st=287]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=288 [st=288]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=288 [st=288]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=289 [st=289]    HTTP_CACHE_WRITE_DATA  [dt=4]
t=293 [st=293]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=293 [st=293]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=294 [st=294]    HTTP_CACHE_WRITE_DATA  [dt=1]
t=295 [st=295]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=295 [st=295]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=296 [st=296]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=296 [st=296]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=296 [st=296]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=297 [st=297]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=297 [st=297]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=297 [st=297]    HTTP_TRANSACTION_READ_BODY  [dt=1]
t=298 [st=298]    HTTP_CACHE_WRITE_DATA  [dt=3]
t=301 [st=301]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 32768
t=301 [st=301]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=301 [st=301]    HTTP_CACHE_WRITE_DATA  [dt=3]
t=305 [st=305]    URL_REQUEST_JOB_FILTERED_BYTES_READ
                  --> byte_count = 5390
t=305 [st=305]    HTTP_TRANSACTION_READ_BODY  [dt=0]
t=305 [st=305]    HTTP_CACHE_WRITE_DATA  [dt=0]
t=305 [st=305] -REQUEST_ALIVE
</pre>
</details>

but request for the favicon is blocked:
<details>
<pre>
3599: URL_REQUEST
https://cloud.githubusercontent.com/favicon.ico
Start Time: 2017-04-27 20:19:13.326

t=366 [st=0] +REQUEST_ALIVE  [dt=4]
              --> priority = "MEDIUM"
              --> url = "https://cloud.githubusercontent.com/favicon.ico"
t=367 [st=1]   +URL_REQUEST_DELEGATE  [dt=3]
t=367 [st=1]      DELEGATE_INFO  [dt=3]
                  --> delegate_blocked_by = "extension uBlock Origin"
t=370 [st=4]      CHROME_EXTENSION_ABORTED_REQUEST
                  --> extension_id = "iphimjoabppgpifmkfadpbjigcefepdp"
t=370 [st=4]   -URL_REQUEST_DELEGATE
t=370 [st=4]    CANCELLED
                --> source = "delegate"
t=370 [st=4]   +URL_REQUEST_START_JOB  [dt=0]
                --> load_flags = 33024 (MAYBE_USER_GESTURE | VERIFY_EV_CERT)
                --> method = "GET"
                --> url = "https://cloud.githubusercontent.com/favicon.ico"
t=370 [st=4]   -URL_REQUEST_START_JOB
                --> net_error = -20 (ERR_BLOCKED_BY_CLIENT)
t=370 [st=4]    URL_REQUEST_DELEGATE  [dt=0]
t=370 [st=4] -REQUEST_ALIVE
              --> net_error = -20 (ERR_BLOCKED_BY_CLIENT)
</pre>
</details>

And it's interesting to see the difference: `CHROME_EXTENSION_ABORTED_REQUEST` and `CANCELLED` appear when actual blocking is happening. So maybe `delegate_blocked_by` appears regardless of the blocking happening, maybe it's like a description of what's trying to check if it wants to block or not. [source code](https://chromium.googlesource.com/chromium/src/net/+/master/url_request/url_request.cc)

**Comment by gorhill** [comment]
uMatrix uses a placeholder image for blocked images. Chromium is seeing that placeholder image, seeing that it is quite small and thus no need to zoom in. In the end, that is just expected from using uMatrix: blocking stuff may break pages, and unblocking may be necessary to unbreak page. That is one such case here.

**Comment by xftroxgpx** [comment]
Roger that :)

**Comment by gorhill** [comment]
Just to be clear about what is going on here, a similar issue was opened somewhere in uBO or uMatrix a long time ago.

- Chrome API request the image as a document, `main_frame` internally.
- `main_frame` are not blocked by uMatrix unless the hostname itself is blocked.
- So the image is correctly fetched by the browser
- The browser create an DOM from thin air and put the downloaded image in it.
- uMatrix's content script looks for blocked image, to replace them with its placeholder.
- uMatrix finds the the image, and evaluate whether it was blocked. It find it was blocked when feeding its URL to uMatrix filtering engine.
    - Except that in reality it was not blocked, because it was downloaded as `main_frame`, not as `image`.
    - But the content script does not know that, it's outside of its knowledge, so it uses the placeholder.
- All this will be made more obvious if one choose the enable "Collapse placeholder of blocked elements".

So TL;DR: uMatrix's content script does not know the image was downloaded as `main_frame` and should not be blocked. It's a quirky edge case of the chrome API -- there is not link between webRequest API and DOM elements.

**Comment by xftroxgpx** [comment]
Thank you gorhill for the explanation. This seems to be the earlier report you mentioned: https://github.com/gorhill/uBlock/issues/2076#issuecomment-254194734
